src/qibo/gates/abstract.py

```diff
@@ -70,7 +83,35 @@ def to_json(self):
         for value in encoded_simple:
             if isinstance(encoded[value], set):
                 encoded_simple[value] = list(encoded_simple[value])
-        return json.dumps(encoded_simple)
+
+        encoded_simple["_class"] = type(self).__name__
+
+        return encoded_simple
+
+    @staticmethod
+    def load(raw: dict):
```

Comment:
Maybe `from_dict` instead?

Reply:
This is fine, but `load` is just simpler than `from_dict`. Since there was no `to_dict` I preferred the general name.